File: app/src/main/java/br/com/hussan/githubapi/AppModule.java
package br.com.hussan.githubapi;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by hussan on 10/10/16.
 */
@Module
public class AppModule {

    GithubApp githubApp;

    public AppModule(GithubApp githubApp) {
        this.githubApp = githubApp;
    }

    @Singleton
    @Provides
    public GithubApp providesGithutApp()
    {
        return githubApp;
    }
}

File: app/src/main/kotlin/br/com/hussan/githubapi/AppModule.kt
package br.com.hussan.githubapi

import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * Created by hussan on 10/10/16.
 */
@Module
class AppModule(internal var githubApp: GithubApp) {

    @Singleton
    @Provides
    fun providesGithutApp(): GithubApp {
        return githubApp
    }
}
